### PR TITLE
Partitioning

### PR DESCRIPTION
**Pull request by joshwie**

### CODE REVIEW

File: Backend/scala-backend/test/models/PartitioningAlgoTest.scala [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+      {
+        for (j <- 0 until nrSus) {
+          if (i != j && adjacencyMatrix(i)(j)) {
+            edges.addOne((i, j))
+          }
+        }
+      }
+    }
+    edges.toArray
+  }
```
Comment (rauwuckl) [inline Backend/scala-backend/test/models/PartitioningAlgoTest.scala:95]: add unit test for the countEdges function

File: Backend/scala-backend/test/models/PartitioningAlgoTest.scala [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+}
```
Comment (rauwuckl) [inline Backend/scala-backend/test/models/PartitioningAlgoTest.scala:96]: add edge case test of empty class

**Review by rauwuckl — CHANGES_REQUESTED** [review]

**Review by rauwuckl — COMMENTED** [review]
ignore this

**Review by rauwuckl — APPROVED** [review]
tipitop
